### PR TITLE
Use np.array instead of list #121

### PR DESCRIPTION
**Pull request by AshrafAlmohagryDeltares**
I have updated the prev_sol to use np.array instead of list and I updated the testing. Please review it.

### CODE REVIEW

File: src/omotes_simulator_core/solver/network/assets/base_item.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ def __init__(self, number_of_unknowns: int, name: str, _id: str, number_connecti
         self.number_of_unknowns = number_of_unknowns
         self.number_of_connection_point = number_connection_points
         self.matrix_index = 0
-        self.prev_sol: list[float] = [0.0] * self.number_of_unknowns
+        self.prev_sol = np.zeros(self.number_of_unknowns)
```
Comment (vanmeerkerk) [inline src/omotes_simulator_core/solver/network/assets/base_item.py:39]: You need to include prior to the __init__ that the prev_sol is of the type np.ndarray.

`

class BaseItem(ABC):
    """A base class for items in a network."""

   prev_sol: np.ndarray
    """The previous solution of the calculation, defaults to an array of zeros with length (number of unknowns,)

    def __init__(self, number_of_unknowns: int, name: str, _id: str, number_connection_points: int):
`

**Review by vanmeerkerk — CHANGES_REQUESTED** [review]
Minor change required see my comment.

File: src/omotes_simulator_core/solver/network/assets/base_item.py (hunk 1 above)
Comment (AshrafAlmohagryDeltares) [reply src/omotes_simulator_core/solver/network/assets/base_item.py:43]: I have included it.

**Review by vanmeerkerk — APPROVED** [review]
Checked

**Pull request merged**